src/constraint/encoding.ts

```diff
@@ -32,11 +32,12 @@ export class EncodingConstraintModel extends AbstractConstraintModel {
       const hasRequiredPropertyAsEnumSpec = some(
         this.constraint.properties,
         function(prop) {
-          if (hasNestedProperty(prop)) {
-            let nestedEncProp = getNestedEncodingProperty(prop);
-            let parent = nestedEncProp.parent;
-            let child = nestedEncProp.child;
-            return isEnumSpec(encQ[parent][child]);
+          const nestedEncProp = getNestedEncodingProperty(prop);
+          if (nestedEncProp) {
+            if (nestedEncProp.parent === prop) {
```

Comment:
Why should `parent === prop`?    Have you really use to debugger to test this?

Look at what are values of `prop`, `parent`, `child`.